src/createTokenBridgePrepareTransactionRequest.ts

```diff
@@ -0,0 +1,86 @@
+import { Address, PublicClient, encodeFunctionData } from 'viem';
+
+import { tokenBridgeCreator } from './contracts';
+import {
+  createTokenBridgeDefaultGasLimit,
+  createTokenBridgeDefaultRetryablesFees,
+} from './constants';
+import { validParentChainId } from './types/ParentChain';
+import { createTokenBridgeGetInputs } from './createTokenBridge-ethers';
+import { publicClientToProvider } from './ethers-compat/publicClientToProvider';
+import { isCustomFeeTokenChain } from './utils/isCustomFeeTokenChain';
+import {
+  TransactionRequestGasOverrides,
+  TransactionRequestRetryableGasOverrides,
+  applyGasOverrides,
+} from './utils/gasOverrides';
+
+export async function createTokenBridgePrepareTransactionRequest({
+  params,
+  parentChainPublicClient,
+  orbitChainPublicClient,
+  account,
+  gasOverrides,
+  retryableGasOverrides,
+}: {
+  params: { rollup: Address; rollupOwner: Address };
+  parentChainPublicClient: PublicClient;
+  orbitChainPublicClient: PublicClient;
+  account: Address;
+  gasOverrides?: TransactionRequestGasOverrides;
+  retryableGasOverrides?: TransactionRequestRetryableGasOverrides;
+}) {
+  const chainId = parentChainPublicClient.chain?.id;
+
+  if (!validParentChainId(chainId)) {
+    throw new Error('chainId is undefined');
+  }
+
+  const parentChainProvider = publicClientToProvider(parentChainPublicClient);
+  const orbitChainProvider = publicClientToProvider(orbitChainPublicClient);
+
+  const { inbox, maxGasForContracts, gasPrice, retryableFee } = await createTokenBridgeGetInputs(
+    account,
+    parentChainProvider,
+    orbitChainProvider,
+    tokenBridgeCreator.address[chainId],
+    params.rollup,
+  );
+
+  const chainUsesCustomFee = await isCustomFeeTokenChain({
+    rollup: params.rollup,
+    parentChainPublicClient,
+  });
+
+  const request = await parentChainPublicClient.prepareTransactionRequest({
+    chain: parentChainPublicClient.chain,
+    to: tokenBridgeCreator.address[chainId],
+    data: encodeFunctionData({
+      abi: tokenBridgeCreator.abi,
+      functionName: 'createTokenBridge',
+      args: [inbox, params.rollupOwner, maxGasForContracts, gasPrice],
+    }),
+    value: chainUsesCustomFee ? 0n : retryableFee,
+    account: account,
+  });
+
+  // potential gas overrides (gas limit)
+  if (gasOverrides && gasOverrides.gasLimit) {
+    request.gas = applyGasOverrides({
+      gasOverrides: gasOverrides.gasLimit,
+      estimatedGas: request.gas,
+      defaultGas: createTokenBridgeDefaultGasLimit,
+    });
+  }
```

Comment:
I think we can make this a bit simpler (combined with the change from the other comment):

```suggestion
  // potential gas overrides (gas limit)
  if (gasOverrides && gasOverrides.gasLimit) {
    request.gas = applyPercentIncrease({
      base: gasOverrides.gasLimit.base ?? request.gas,
      percentIncrease: gasOverrides.gasLimit.percentIncrease,
    });
  }
```

Don't think we really need to have defaults in here, it's good enough to expose as constants so the consumer can do:

```
createTokenBridge({
  ...,
  gasOverrides: {
    gasLimit: {
      base: createTokenBridgeDefaultGasLimit
    }
  }
})
```